tls/s2n_cipher_suites.h

```diff
@@ -160,6 +160,7 @@ extern struct s2n_cipher_suite s2n_tls13_chacha20_poly1305_sha256;
 
 extern int s2n_cipher_suites_init(void);
 extern int s2n_cipher_suites_cleanup(void);
+S2N_RESULT s2n_iana_to_cipher_suite(const uint8_t iana[], struct s2n_cipher_suite **cipher_suite);
```

Comment:
On the next line the data type is defined as `uint8_t[S2N_TLS_CIPHER_SUITE_LEN]` but here it is just `uint8_t[]`. I don't massively care which we do, but we should stay consistent.

(As a nitpick, I think we should be using `uint16_t` for these values anyway. The more I think about this the more uncomfortable I am with this construction. I feel that we should either be using a `uint16_t` or a dedicated struct for type safety.
```C
struct iana_suite_id {
    uint8_t id[S2N_TLS_CIPHER_SUITE_LEN]
};
```
)

Reply:
`uint8_t iana[S2N_TLS_CIPHER_SUITE_LEN]` is equivalent to `uint8_t iana[]` because C, but I can add the length for documentation / consistency.

This method should probably take an array as long as we're [representing the iana as an array on cipher suites](https://github.com/aws/s2n-tls/blob/main/tls/s2n_cipher_suites.h#L93), and changing cipher suites is out of scope here. Want me to create an issue though?